### PR TITLE
osd: make the HEALTH_WARN detail less confusing when NOOUT is set

### PR DESCRIPTION
**Pull request by gregsfortytwo**
This warning is very, very misleading. So, change it to be an OR instead of a set, to be at least somewhat better.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
I found this *very* confusing when I saw it in a log recently — I am inclined to put this fix everywhere we can. But if there are external users watching for it, this will require updating their scripts, so maybe we shouldn't?

I didn't see this string present anywhere else in the code, so we're probably not going to fail any tests as a result of changing this string. But I haven't even compiled it yet — consider this PR your discussion starter, RADOS team. :)

File: src/osd/OSDMap.cc [hunk 1]
```diff
@@ -7039,7 +7039,7 @@ void OSDMap::check_health(CephContext *cct,
     }
     if (!detail.empty()) {
       ostringstream ss;
-      ss << detail.size() << " OSDs or CRUSH {nodes, device-classes} have {NOUP,NODOWN,NOIN,NOOUT} flags set";
+      ss << detail.size() << " OSDs or CRUSH {nodes, device-classes} have {NOUP|NODOWN|NOIN|NOOUT} flags set";
```
Comment (rzarzynski) [inline src/osd/OSDMap.cc:7042]: Agreed, the commas are confusing to say the least. Technically I'm fine with the patch but it needs to be properly communicated (CDM + mail to ceph-users?) IMHO.

**Review by rzarzynski — APPROVED** [review]

File: src/osd/OSDMap.cc (hunk 1 above)
Comment (athanatos) [reply src/osd/OSDMap.cc:7042]: Mmm, I think we can go even more explicit:

```
" There are OSDs or other crush CRUSH nodes have at least one of NOUP, NODOWN, NOIN, or NOOUT flags set"

**Review by athanatos — APPROVED** [review]

File: src/osd/OSDMap.cc (hunk 1 above)
Comment (xiexingguo) [reply src/osd/OSDMap.cc:7042]: > 

+1

**Pull request closed without merging**